pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -0,0 +1,835 @@
+			return "", err
+		}
+
+		// Create a ClusterRoleBinding
+		clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: bootstraputil.BuildBootstrapperName(name),
+			},
+		}
+		if _, err := controllerutil.CreateOrUpdate(ctx, a.gardenClient.Client(), clusterRoleBinding, func() error {
+			clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     bootstraputil.GardenerSeedBootstrapper,
+			}
+			clusterRoleBinding.Subjects = []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      saName,
+					Namespace: v1beta1constants.GardenNamespace,
+				},
+			}
+			return nil
+		}); err != nil {
+			return "", err
+		}
+
+		// Get bootstrap kubeconfig from service account secret
+		bootstrapKubeconfig, err = bootstraputil.MarshalKubeconfigWithToken(&restConfig, string(saSecret.Data[corev1.ServiceAccountTokenKey]))
+		if err != nil {
+			return "", err
+		}
+
+	case seedmanagementv1alpha1.GardenConnectionBootstrapToken:
+		// Create bootstrap token with bootstrap kubeconfig in order to create CSR
+
+		// Get bootstrap token secret
+		tokenID := utils.ComputeSHA256Hex([]byte(name))[:6]
+		secret := &corev1.Secret{}
+		if err := a.gardenClient.Client().Get(ctx, kutil.Key(metav1.NamespaceSystem, bootstraptokenutil.BootstrapTokenSecretName(tokenID)), secret); client.IgnoreNotFound(err) != nil {
+			return "", err
+		}
+
+		// Refresh bootstrap token if needed
+		refreshBootstrapToken := true
+		var bootstrapTokenSecret *corev1.Secret
+		if expirationTime, ok := secret.Data[bootstraptokenapi.BootstrapTokenExpirationKey]; ok {
+			t, err := time.Parse(time.RFC3339, string(expirationTime))
+			if err != nil {
+				return "", err
+			}
+			if !t.Before(metav1.Now().UTC()) {
+				refreshBootstrapToken = false
+				bootstrapTokenSecret = secret
+			}
+		}
+		if refreshBootstrapToken {
+			bootstrapTokenSecret, err = kutil.ComputeBootstrapToken(ctx, a.gardenClient.Client(), tokenID, fmt.Sprintf("A bootstrap token for the Gardenlet for shooted seed %q.", name), 24*time.Hour)
+			if err != nil {
+				return "", err
+			}
+		}
+
+		// Get bootstrap kubeconfig from bootstrap token
+		bootstrapKubeconfig, err = bootstraputil.MarshalKubeconfigWithToken(&restConfig, kutil.BootstrapTokenFrom(bootstrapTokenSecret.Data))
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return string(bootstrapKubeconfig), nil
+}
+
+func (a *actuator) prepareSeedConfig(ctx context.Context, managedSeed *seedmanagementv1alpha1.ManagedSeed, shoot *gardencorev1beta1.Shoot, seedConfig *configv1alpha1.SeedConfig) (*configv1alpha1.SeedConfig, error) {
+	// Initialize the seed config if it's nil
+	if seedConfig == nil {
+		seedConfig = &configv1alpha1.SeedConfig{}
+	}
+
+	// Set the seed name to the managed seed name
+	seedConfig.Seed.Name = managedSeed.Name
```

Comment:
Why do we do this? I think we should use the name specified in the GardenletConfiguration (which should ideally be the same but not necessarily). 
This is also why I think that displaying the Seed name [here](https://github.com/gardener/gardener/blob/fe0d633e505050144240d29c456f2ecf4b29ec3f/pkg/registry/seedmanagement/managedseed/storage/tableconvertor.go#L79) makes sense.

Reply:
See my comment above.

Reply:
Mhm sorry, I do not know which comment you refer to / cannot find it.

Reply:
The following comment to `tableconvertor.go`:

> @danielfoehrKn I would prefer to forbid specifying the Seed name explicitly and always set it to the ManagedSeed name (or a predictable, unique derivative). Otherwise we would have to deal with naming conflicts, etc. Yes, you are right it should be documented and validated, I just haven't got to that point.

Reply:
thanks. Makes sense. Also fine with me.